services/retentionmgr/metadataDep.go

```diff
@@ -130,19 +130,22 @@ func (t *metadataDepImpl) GetExtents(destID destinationID) (extents []*extentInf
 		for _, extStats := range resp.GetExtentStatsList() {
 
 			extent := extStats.GetExtent()
+
 			storeUUIDs := extent.GetStoreUUIDs()
+			storehosts := make([]storehostID, 0, len(storeUUIDs))
+
+			for _, storeUUID := range storeUUIDs {
+				storehosts = append(storehosts, storehostID(storeUUID))
+			}
 
 			extInfo := &extentInfo{
 				id:                 extentID(extent.GetExtentUUID()),
 				status:             extStats.GetStatus(),
 				statusUpdatedTime:  time.Unix(0, extStats.GetStatusUpdatedTimeMillis()*int64(time.Millisecond)),
-				storehosts:         make([]storehostID, 0, len(storeUUIDs)),
+				storehosts:         storehosts,
```

Comment:
You cannot typecast slices (container types, in general) with differing containing types -- afaik, the only way to convert is to do the copy. Yes, I know it sucks. :-)